### PR TITLE
Make markdown tutorials for `ChatAgent`

### PR DESCRIPTION
**Pull request by jjyaoao**
## Description

Add documentation for core class ChatAgent.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [x] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/get_started/chat_agent.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    ```
+
+8. **Submitting an External Message as an Assistant Response**:
+    The `submit_message` method allows you to provide an external message as if it were a response from the chat LLM.
+    ```python
+    external_message = BaseMessage(...)
+    agent.submit_message(external_message)
+    ```
+
+9. **Performing a Chat Session Step**:
```
Comment (dandansamax) [inline docs/get_started/chat_agent.md:124]: This is the most important member function of ChatAgent for most of the users. Please put it on the top of the file.

File: docs/get_started/chat_agent.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    ```python
+    stream_response = {...}
+    prompt_tokens_count = 100
+    messages, reasons, usage, response_id = agent.handle_stream_response(response=stream_response, prompt_tokens=prompt_tokens_count)
+    print(messages)
+    >>> [<BaseMessage content="Streamed Message 1">, <BaseMessage content="Streamed Message 2">]
+    ```
+
+14. **Handling Token Exceedance**:
+    The `step_token_exceed` method returns a response containing the number of tokens and information about called functions when the number of tokens exceeds the set limit.
```
Comment (dandansamax) [inline docs/get_started/chat_agent.md:169]: Actually it should be a private function. Could you please just remove it?

File: docs/get_started/chat_agent.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+agent = ChatAgent(
+    system_message=BaseMessage(...),
+    model=ModelType.GPT_3_5_TURBO,
+    message_window_size=5,
+    output_language="en"
+)
+```
+
+## Understanding the Properties of the `ChatAgent` Class
+
```
Comment (dandansamax) [inline docs/get_started/chat_agent.md:39]: It will be better to add properties' type in this introduction.

**Review by dandansamax — CHANGES_REQUESTED** [review]
Hi. Thank you so much for working on documents and we appreciate your contribution. The document looks clear. Here are Just some advises.

In my opinion, the documents in the `get_started` section should introduce the module as simple, stupid as possible. We just write something for new users to quickly understand the basic idea of the module and to get started. The detailed information can be checked in the API docs and the code. Thus, we need more intuitive description with examples in a real productive situation.

File: docs/get_started/chat_agent.md (hunk 1 above)
Comment (jjyaoao) [reply docs/get_started/chat_agent.md]: > This is the most important member function of ChatAgent for most of the users. Please put it on the top of the file.

done

File: docs/get_started/chat_agent.md (hunk 2 above)
Comment (jjyaoao) [reply docs/get_started/chat_agent.md]: > Actually it should be a private function. Could you please just remove it?

done

File: docs/get_started/chat_agent.md (hunk 3 above)
Comment (jjyaoao) [reply docs/get_started/chat_agent.md:39]: > It will be better to add properties' type in this introduction.

done

**Review by Obs01ete — APPROVED** [review]
This tutorials are very valuable as they are. Let's merge them. Later we will rework the tutorials to be more simplistic and less technical.

**Comment by lightaime** [comment]
Thanks @jjyaoao for the PR. I will look into it soon!

@Obs01ete There are some small flaws. Please wait for my review.

**Comment by lightaime** [comment]
Thanks @jjyaoao. I close this for now since it is outdated.

**Pull request closed without merging**